app/models/concerns/eds_subjects.rb

```diff
@@ -0,0 +1,99 @@
+module EdsSubjects
+  def eds_subjects
+    @eds_subjects ||= Subject.from(self['eds_subjects'])
+  end
+
+  def eds_subjects_geographic
+    @eds_subjects_geographic ||= Subject.from(self['eds_subjects_geographic'])
+  end
+
+  def eds_subjects_person
+    @eds_subjects_person ||= Subject.from(self['eds_subjects_person'])
+  end
+
+  def eds_author_supplied_keywords
+    @eds_author_supplied_keywords ||= Subject.from(self['eds_author_supplied_keywords'])
+  end
+
+  # EDS-centric subject with linking capability
+  class Subject
+    attr_reader :terms
+    delegate :present?, to: :terms
+
+    class Term < Struct.new(:fieldCode, :searchTerm, :label)
+      delegate :to_s, to: :label
+
+      def to_html
+        field = case fieldCode
```

Comment:
Do you think we can break it down a bit?  It looks like there are a few things this method is doing that could be broken down into smaller responsibilities.